src/backdoor.rs

```diff
@@ -18,20 +18,20 @@ pub(crate) const COMMAND_ERPC: u32 = 0x1E;
 /// Try to acquire access to the backdoor, but do NOT probe its presence.
 ///
 /// On Linux, this requires running with `CAP_SYS_RAWIO` privileges.
-pub fn access_backdoor() -> Result<BackdoorGuard, VmwError> {
-    BackdoorGuard::change_io_access(true)?;
+pub fn access_backdoor(ignore_access_errors: bool) -> Result<BackdoorGuard, VmwError> {
+    let changed = BackdoorGuard::change_io_access(true, ignore_access_errors)?;
     Ok(BackdoorGuard {
-        release_on_drop: true,
+        release_on_drop: changed,
     })
 }
 
 /// Try to acquire access to the backdoor, and probe its presence.
 ///
 /// On Linux, this requires running with `CAP_SYS_RAWIO` privileges.
-pub fn probe_backdoor() -> Result<BackdoorGuard, VmwError> {
-    BackdoorGuard::change_io_access(true)?;
+pub fn probe_backdoor(ignore_access_errors: bool) -> Result<BackdoorGuard, VmwError> {
```

Comment:
I would say a separate `probe_backdoor_optional` method is cleaner than passing a `bool`.  Take for example [open_file_optional](https://github.com/cgwalters/openat-ext/blob/64ca702d60308aabbd220a724f6f789ef6d99606/src/lib.rs#L31) vs `open_file(true)` or so.

Boolean arguments are just harder to read because you need to go to the call site to understand what they're doing.  And it's easy to get them backwards; is the boolean saying to ignore errors or to return them, etc.

Reply:
(Now internally in the code they can often end up calling a private helper that takes a bool, but the goal is to avoid confusion for people calling the method)

Reply:
Ah, good suggestion, the `_optional` methods would fit nice here. I'll rework in that direction.